docs/cswinrt3.0-spec.md

```diff
@@ -0,0 +1,129 @@
+
+### Other breaking changes
+
+The API surface of projected runtime classes will be significantly streamlined compared to CsWinRT 2.x. In particular:
+- **`IWinRTObject` and all related infrastructure in each projected runtime class will be removed.** All the shared functionality will be provided by the base `WindowsRuntimeObject` class, which will also handle `IDynamicInterfaceCastable` caching. This also reduces the object size for each runtime class type, as well as improving binary size.
+- **`As<I>()` will be removed.** This functionality is available via `WindowsRuntimeActivationFactory`.
+- **`FromAbi(nint)` will be removed.** This functionality is available via the marshaller type for each runtime class type.
+- **All `IEquatable<T>` support will be removed.** Projected runtime class instances will compare via reference equality, like any other object. Equality support for the underlying native object will be provided as opt-in via the `WindowsRuntimeMarshal` type.
+
+## Adoption plan
+
+While in general the projected WinRT API surface will be source compatible, CsWinRT 3.0 will not be binary compatible with previous versions. Meaning that to switch to CsWinRT 3.0 in a given project, all WinRT projections will need to be updated to reference CsWinRT 3.0 as well (which will update the generated projections in that project). This will be somewhat similar (although more drastic) to the change to CsWinRT 2.x, which while more backwards compatible, still required projections to be updated to avoid .dll conflicts. We will continue servicing CsWinRT 2.x after releasing CsWinRT 3.0, for people that can't migrate to it yet, or for those that rely on features that will no longer be supported (such as embedded mode). We will then work on a deprecation plan for 2.x, to apply after this transition period where CsWinRT 2.x will continue receiving critical bug fixes.
+
+### Multi-targeting support
+
+Because CsWinRT 3.0 is fundamentally incompatible with CsWinRT 2.x, we need a way for NuGet package authors to keep supporting CsWinRT 2.x in parallel with CsWinRT 3.0, so that developers that can't upgrade their apps and libraries to CsWinRT 3.0 (e.g. because they're blocked by one or more dependencies not having updated just yet) will be able to still get new updates for those packages. The solution for this is to leverage the multi-targeting feature from NuGet. Specifically, the .NET SDK will add support for multi-targeting the CsWinRT version based on the revision number of the Windows SDK version indicated in the TFM (see [dotnet/sdk#49721](https://github.com/dotnet/sdk/pull/49721)).
+
+Using a revision value of `.1` will include CsWinRT 3.0 and enable all its tooling. A revision of `.0` will use CsWinRT 2.x, e.g.:
+- `net10.0-windows10.0.22621.0` ---> CsWinRT 2.x
+- `net10.0-windows10.0.22621.1` ---> CsWinRT 3.0
```

Comment:
Nit: Wonder if you should leave a bigger gap there to give each framework some breathing room for their own unplanned revisions. Like perhaps `net10.0-windows10.0.22621.1000`.

Otherwise you could end up in a situation where we have targets .0, .1, and .2 which would be mega confusing.

Reply:
I think @riverar's point has merit, the fact .NET/SDK nuget owners do not apparently already do this today only re-enforces that argument. This would be to account for unplanned expected revisions that have to be made, I see a world where this would be a thing that has to occur for critical bug fixes and indeed, not having breathing room for supporting that would not enable this to work properly.

As for whenever its being done already or not, I think the answer is yes, take a look at this ref nuget package for example, there are indeed revisions binding to the windows OS revisions themselves in here, and we had sdk updates in the past too for the same baseline windows in market stable release: https://www.nuget.org/packages/Microsoft.Windows.SDK.NET.Ref